plugins/omazureeventhubs/omazureeventhubs.c

```diff
@@ -1773,3 +1037,341 @@ pn_timestamp_t time_now(void)
   if (gettimeofday(&now, NULL)) return 0; 
   return ((pn_timestamp_t)now.tv_sec) * 1000 + (now.tv_usec / 1000);
 }  
+
+/*
+*	Start PROTON Handling Thread
+*/
+static rsRetVal proton_run_thread(instanceData *pData)
+{
+	DEFiRet;
+	int iErr = 0;
+	if (	protonWrkrInfo == NULL ||
+		!protonWrkrInfo->bThreadRunning) {
+		DBGPRINTF("Create Proton Protocol Thread for %p\n", pData);
+		protonWrkrInfo = calloc(1, sizeof(struct protonWrkrInfo_s));
+		if (protonWrkrInfo == NULL) {
+			LogError(errno, RS_RET_OUT_OF_MEMORY, "omazureeventhubs: proton_run_thread allocation failed.");
+			ABORT_FINALIZE(RS_RET_OUT_OF_MEMORY);
+		}
+
+		/* init worker info structure! */
+		protonWrkrInfo->pInstData = pData; /* Set reference pointer */
+		do {
+			iErr = pthread_create(&protonWrkrInfo->tid, NULL, proton_thread, &(protonWrkrInfo));
+			if (!iErr) {
+				protonWrkrInfo->bThreadRunning = 1;
+				FINALIZE;
+			}
+		} while (iErr == EAGAIN);
+	} else {
+		DBGPRINTF("Proton Protocol Thread (tid %d) already running\n", protonWrkrInfo->bThreadRunning);
+	}
+finalize_it:
+	if (iRet != RS_RET_OK) {
+		LogError(0, RS_RET_SYS_ERR, "omazureeventhubs: proton_run_thread thread create failed with error: %d", 
+			iErr);
+	}
+	RETiRet;
+}
+/*	Stop PROTON Handling Thread
+*/
+static rsRetVal 
+proton_shutdown_thread(instanceData *pData)
+{
+	DEFiRet;
+	if (	protonWrkrInfo != NULL && 
+		protonWrkrInfo->bThreadRunning) {
+		DBGPRINTF("STOPPING Proton Protocol Thread for %p\n", pData);
+		int r = pthread_cancel(protonWrkrInfo->tid);
+		if(r == 0) {
+			pthread_join(protonWrkrInfo->tid, NULL);
+		}
+		DBGPRINTF("STOPPED Proton Protocol Thread for %p\n", pData);
+		free(protonWrkrInfo);
+		protonWrkrInfo = NULL;
+	}
+	FINALIZE;
+finalize_it:
+	RETiRet;
+}
+
+/*
+*	Workerthread function for a single ProActor Handler
+ */
+static void *
+proton_thread(void __attribute__((unused)) *myInfo)
+{
+	// protonWrkrInfo_s *thisWrkrInfo = (struct protonWrkrInfo*) myInfo;
+	instanceData *const pData = (instanceData *const) protonWrkrInfo->pInstData;
+
+	DBGPRINTF("proton_thread started protocol workerthread for %p:%s:%s/%s\n", 
+		pData, pData->azurehost, pData->azureport, pData->container); 
+
+	do {
+		if (	pData->pnProactor != NULL) {
+			// Process Protocol events
+			pn_event_batch_t *events = pn_proactor_wait(pData->pnProactor);
+			pn_event_t *event;
+			for (event = pn_event_batch_next(events); event; event = pn_event_batch_next(events)) {
+				handleProton(pData, event);
+			}
+			pn_proactor_done(pData->pnProactor, events);
+		}
+
+		if(glbl.GetGlobalInputTermState() == 1)
+			break; /* terminate input! */
+
+		/* Note: the additional 10000ns wait is vitally important. It guards rsyslog
+		 * against totally hogging the CPU if the users selects a polling interval
+		 * of 0 seconds. It doesn't hurt any other valid scenario. So do not remove.
+		 * rgerhards, 2008-02-14
+		 */
+		if(glbl.GetGlobalInputTermState() == 0)
+			srSleep(0, 100000);
```

Comment:
is this whole section really needed or a "copy victim" form a "copy victim" that copied imfile? Can this code really hog the CPU? I see the pn_proactor-wait() call plus myself insight the comment, which smells fishy ;-)

Reply:
copy victim but needs to be confirmedd if needed.

Reply:
question is: is it possible for azure to hammer the server? Doesn't look so. Pls change at least the comment (I didn't do it ;-)) and note that this is fishy and needs to be reviewed later.

Reply:
Actually that whole block can be removed. (will be next amend)